# Installation

Install the following libraries are available

	install.packages("gplots", dependencies=TRUE)

	install.packages("ggplot2", dependencies=TRUE)

	install.packages("sjPlot", dependencies=TRUE)

	install.packages("dynamicTreeCut", dependencies=TRUE)

Install topGO and edgeR from bioconductor

## Local installation

	mkdir ~/git

	cd ~/git

	git clone https://GitHub.com/parisveltsos/dpse_mating.git

	cd ~/git/dpse_mating

	mkdir output

The output folder contents can be generated by running the scripts. An archive of the output is available for download at [osf](https://osf.io/z7fm9/?view_only=054171ba3f534f839a0814fa1b8f9f61).

The correspondence between tabs in File S3 and output files are shown below

	testis EvsM - ~/git/dpse_mating/output/dedata/testis/GO_0.05_maleVirgins_E.M/
	agland EvsM - ~/git/dpse_mating/output/dedata/agland/GO_0.05_maleVirgins_E.M/
	FRT EvsM - ~/git/dpse_mating/output/dedata/rtract/GO_0.05_EvsM/
	ov EvsM - ~/git/dpse_mating/output/dedata/ovary/GO_0.05_EvsM/
	Main effect of SS line FRT - ~/git/dpse_mating/output/dedata/rtract/GO_0.05_allEvsM/
	Main effect of mating FRT - ~/git/dpse_mating/output/dedata/rtract/GO_0.05_E.MvsEE/
	line x mating FRT - ~/git/dpse_mating/output/dedata/rtract/GO_0.05_E.MMvsM.EE/
	Main effect of SS line OV - ~/git/dpse_mating/output/dedata/ovary/GO_0.05_allEvsM/
	Main effect of mating OV - ~/git/dpse_mating/output/dedata/ovary/GO_0.05_E.MvsEE/
	Main female effect FRT - ~/git/dpse_mating/output/dedata/rtract/GO_0.05_EE.EMvsMM.ME/
	Main male effect FRT - ~/git/dpse_mating/output/dedata/rtract/GO_0.05_EE.MEvsEM.MM/
	Main female effect OV - ~/git/dpse_mating/output/dedata/ovary/GO_0.05_EE.EMvsMM.ME/
	male x female OV - ~/git/dpse_mating/output/dedata/ovary/GO_0.05_EE.MMvsEM.ME/
	venn FRT M - ~/git/dpse_mating/output/venn/frt.M*
	venn FRT E - ~/git/dpse_mating/output/venn/frt.E*
	venn 32 - ~/git/dpse_mating/output/venn/venn32GO
	
# Analyses (in order)

## Main DE analysis

Note, more output than used in the paper is produced. The script 

	* performs DE analysis and reports significant genes along with their log2FC and FDR
	* plots MDS 
	* plots volcano plot (cpm vs log2FC)
	* performs and plots chisq tests of DE gene number comparisons between different chromosomes, and produces the relevant tables with sjPlot
	* plots the chromosomal distribution of significant genes
	* makes violin plots showing differential expression magnitude in different chromosomes
	* makes heatmaps of the DE genes using all libraries
	
Run the EM contrast analyses for all tissues, with 5% FDR.

	source ~/git/dpse_mating/scripts/batch_analysis.sh

The input files have removed the rDNA genes, and library EEov4, which was an MDS plot outlier.

## GO analysis

Navigate to the folder of the relevant comparisons, and run the analysis. This needs to be done in the terminal for each contrast. eg for testis

	cd ~/git/dpse_mating/output/testis
	
	for i in $(ls | grep GO); do cd $i; source ~/git/dpse_mating/scripts/GO_analysis_05.sh; cd ..; done

The `Fisher.txt` files in the GO_0.05... and GO_DOWN and GO_UP folders make up the GO analysis results ("All DE" "Up" or "Down") summarized in File S3. 

### Venn diagrams

	mkdir -p ~/git/dpse_mating/output/venn/

DE genes from the analysis, eg `~/git/dpse_mating/output/dedata/ovary_matingE_rmEEov4/de_0.05_0_ovary_matingE_rmEEov4.txt` were used as input in [venny](https://bioinfogp.cnb.csic.es/tools/venny/) to generate the dataset in `~/git/dpse_mating/output/venn/venn32GO`. This is analysed as in the GO analysis, above.

The scatterplot.r script below generates the other "venn" data in the output/venn folder, which are analysed as in the GO analysis, above.

## Scatterplot

`~/git/dpse_mating/scripts/scatterplot.r` makes scatterplot panels from Fig 3 and Fig 4.

## Agcp analysis

This analysis produces Figure 2, comparing DE genes from this study with agcp genes from [Karr 2019](https://doi.org/10.1074/mcp.RA118.001098). 

Run `~/git/dpse_mating/scripts/agcpPlot.r` See script for origin of input files.

## Box plots

The data in `input/boxplots` were made from the output of the DE analysis. These were made manually by adding the second column to that indicates the contrast producing it (negative values are the M libraries) from the following files:

For `input/boxplots/female_virgin_boxplotData.txt` the logFC values are from 

* `~/git/dpseMating/output/dedata/ovary/LogCPM_0.05_ovary.txt` 
* `~/git/dpseMating/output/dedata/rtract/LogCPM_0.05_rtract.txt` 

For `input/boxplots/male_virgin_boxplotData.txt` the logFC values are from 

* `~/git/dpseMating/output/dedata/testis/de_0.05_0_testis.txt` 
* `~/git/dpseMating/output/dedata/agland/de_0.05_0_agland.txt` 

Panels A and B for Fig 1 are made with

	`~/git/dpse_mating/scripts/boxplot.r`


Plots were manipulated for presentation in Affinity Designer v1.9.3, mostly to change the location of the gene numbers and change the text of the y axis.
